graph/graph_windows.go

```diff
@@ -155,6 +182,50 @@ func (graph *Graph) TarLayer(img *Image) (arch archive.Archive, err error) {
 		// We keep this functionality here so that we can still work with the VFS
 		// driver during development. VFS is not supported (and just will not work)
 		// for Windows containers.
-		return graph.driver.Diff(img.ID, img.Parent)
+		root := graph.imageRoot(img.ID)
+		mFileName := filepath.Join(root, tarDataFileName)
+		mf, err := os.Open(mFileName)
```

Comment:
It is identical because the windows graph split out duplicated this bit. We
need to get that windows driver into the interface form, then this will
reconcile a lot.
On Jul 16, 2015 21:06, "Stephen Day" <notifications@github.com> wrote:

> In graph/graph_windows.go
> <https://github.com/docker/docker/pull/14067#discussion_r34854738>:
>
> > @@ -155,6 +182,50 @@ func (graph *Graph) TarLayer(img *Image) (arch archive.Archive, err error) {
> >  		// We keep this functionality here so that we can still work with the VFS
> >  		// driver during development. VFS is not supported (and just will not work)
> >  		// for Windows containers.
> > -		return graph.driver.Diff(img.ID, img.Parent)
> > +		root := graph.imageRoot(img.ID)
> > +		mFileName := filepath.Join(root, tarDataFileName)
> > +		mf, err := os.Open(mFileName)
>
> I don't understand why they are split out at this level. There is no
> system specific stuff in here... :(
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/14067/files#r34854738>.
>